apps/ledger-live-mobile/src/screens/SyncOnboarding/SoftwareChecksStep.tsx

```diff
@@ -0,0 +1,402 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { BoxedIcon, Flex, InfiniteLoader, Text } from "@ledgerhq/native-ui";
+import {
+  CircledAlertMedium,
+  CircledCheckSolidMedium,
+} from "@ledgerhq/native-ui/assets/icons";
+import { FlexBoxProps } from "@ledgerhq/native-ui/components/Layout/Flex";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { useGenuineCheck } from "@ledgerhq/live-common/hw/hooks/useGenuineCheck";
+import { useGetLatestAvailableFirmware } from "@ledgerhq/live-common/hw/hooks/useGetLatestAvailableFirmware";
+import { getDeviceModel } from "@ledgerhq/devices";
+import { useTheme } from "styled-components/native";
+
+import GenuineCheckDrawer from "./GenuineCheckDrawer";
+import FirmwareUpdateDrawer from "./FirmwareUpdateDrawer";
+import GenuineCheckCancelledDrawer from "./GenuineCheckCancelledDrawer";
+import UnlockDeviceDrawer from "./UnlockDeviceDrawer";
+import AllowManagerDrawer from "./AllowManagerDrawer";
+
+const softwareStepDelay = 2500;
+const lockedDeviceTimeoutMs = 1000;
+
+type CheckStatus = "inactive" | "active" | "completed" | "failed";
+
+type CurrentSoftwareChecksStep = "none" | "genuine-check" | "firmware-update";
+
+type GenuineCheckUiStepStatus = CheckStatus;
+type GenuineCheckStatus =
+  | "unchecked"
+  | "requested"
+  | "ongoing"
+  | "completed"
+  | "failed";
+type GenuineCheckUiDrawerStatus =
+  | "none"
+  | "requested"
+  | "allow-manager"
+  | "unlock-needed"
+  | "cancelled";
+
+type FirmwareUpdateUiStepStatus = CheckStatus;
+type FirmwareUpdateStatus = "unchecked" | "ongoing" | "completed" | "failed";
+type FirmwareUpdateUiDrawerStatus = "none" | "new-firmware-available";
+
+type CheckCardProps = FlexBoxProps & {
+  title: string;
+  index: number;
+  status: CheckStatus;
+};
+
+const CheckCard = ({ title, index, status, ...props }: CheckCardProps) => {
+  const { colors } = useTheme();
+
+  const getCheckIcon = useCallback(
+    (status: CheckStatus, index: number) => {
+      if (status === "active") {
+        return <InfiniteLoader color={colors.primary.c80} size={24} />;
+      }
+      if (status === "completed") {
+        return <CircledCheckSolidMedium color="success.c100" size={24} />;
+      }
+      if (status === "failed") {
+        return <CircledAlertMedium color="warning.c100" size={24} />;
+      }
+      return <Text variant="body">{index}</Text>;
+    },
+    [colors.primary.c80],
+  );
+
+  return (
+    <Flex flexDirection="row" alignItems="center" {...props}>
+      <BoxedIcon
+        backgroundColor="neutral.c30"
+        borderColor="neutral.c30"
+        variant="circle"
+        Icon={getCheckIcon(status, index)}
```

Comment:
Bad use of this prop, I think this prop expects a Component and here you're passing something like a ReactNode